lib/vehicle.rb

```diff
@@ -6,13 +6,17 @@ class Vehicle
               :make,
               :model,
               :engine
-
+              
+  attr_accessor :registration_date, :plate_type
+              
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registration_date = nil
+    @plate_type = nil
```

Comment:
This works because you are defining the plate type in your facility class. Should it be the facilities job to define it? Or could you write a stand alone method below called `plate_type` where you set each type and then call that method in other classes?